cdap-client/src/main/java/co/cask/cdap/client/MetadataClient.java

```diff
@@ -80,10 +81,28 @@ public MetadataClient(ClientConfig config) {
   public Set<MetadataSearchResultRecord> searchMetadata(Id.Namespace namespace, String query,
                                                         @Nullable MetadataSearchTargetType target)
     throws IOException, UnauthenticatedException {
+    Set<MetadataSearchTargetType> targets = ImmutableSet.of();
+    if (target != null) {
+      targets = ImmutableSet.of(target);
+    }
+    return searchMetadata(namespace, query, targets);
+  }
+
+  /**
+   * Returns search results for metadata.
```

Comment:
"Searches entities in the specified namespace whose metadata matches the specified query"?